lib/xain/helpers.ex

```diff
@@ -37,6 +38,18 @@ defmodule Xain.Helpers do
   def pop!([item | rest]) do
     {rest, item |> Enum.reverse}
   end
+
+  def ensure_valid_contents(contents, _) when is_binary(contents) or is_list(contents) do
+    contents
+  end
+  def ensure_valid_contents(contents, tag_name) when is_number(contents) or is_atom(contents) do
+    Logger.debug "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    to_string(contents)
+  end
+  def ensure_valid_contents(contents, tag_name) do
+    Logger.warn "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    inspect(contents)
```

Comment:
Well, I'm not sure that we should encourage such usage (with implicit to_string)
But I guess, this is not so crucial question. So, I don't mind of your implementation.